### DISCUSSION TITLE
iOS 用json不能运行

### DISCUSSION THREAD

**Discussion by shagfu** [eycorsican/leaf#144] [Q&A]
我的json 内容如下 
let conf = """
{
    "log": {
        "level": "trace"
    },
    "dns": {
        "servers": [
            "223.5.5.5",
            "8.8.8.8"
        ]
    },
    "inbounds": [
        {
            "address": "127.0.0.1",
            "port": 1087,
            "protocol": "http"
        }
    ],
    "outbounds": [
        {
            "protocol": "direct",
            "tag": "direct_out"
        },
        {
            "protocol": "vmess",
            "settings": {
                "address": "30.43.120.182",
                "port": 30005,
                "uuid": "e02431b6-9e5a-11eb-a8b3-0242ac130003"
            },
            "tag": "vmess"
        },
        {
            "protocol": "shadowsocks",
            "settings": {
                "address": "30.43.120.182",
                "method": "aes-256-gcm",
                "password": "123456",
                "port": 30001
            },
            "tag": "shadowsocks_out"
        },
        {
            "protocol": "drop",
            "tag": "drop_out"
        }

    ],
    "rules": [
        {
            "domainKeyword": [
                "baidu"
            ],
            "target": "shadowsocks_out"
        },
        {
            "domain": [
                "www.baidu.com"
            ],
            "target": "shadowsocks_out"
        },
        {
            "external": [
                "site:cn"
            ],
            "target": "direct_out"
        }
    ]
}
然后在将 代码 改成如下
let url = FileManager().containerURL(forSecurityApplicationGroupIdentifier: appGroup)!.appendingPathComponent("running_config.json")
控制台输出 [INFO] inbound listening tcp 127.0.0.1:1087 
请问下json内容对吗？
同样的配置用conf 是可以的。

**Comment by eycorsican** [comment]
没看出什么问题，有监听 HTTP 的日志输出就应该 可以

**Comment by shagfu** [comment]
用conf 配置，输出  tun inbound started,用json 没有输出这条日志，是不是这个函数哪里出错了，json 里面tun-fd = REPLACE-ME-WITH-THE-FD这个 fd需要配置吗？
另外请问下conf http2 如何配置？

**Reply by nahddw** [reply]
老哥 你解決了嗎？iOS如何使用json

**Reply by ixiao** [reply]
我没有用json配置，按照源码的方式，就使用字符串拼接 是可以的

**Reply by ixiao** [reply]
抱歉 我也没有编译。 都是用编辑好的东西

**Reply by 594810899** [reply]
> 抱歉 我也没有编译。 都是用编辑好的东西

我的意思是 libleaf.a这个静态库 你那边是怎么编译生成的，或者是你有没有已经可以使用的 libleaf.a库文件供iOS端使用？

**Reply by ixiao** [reply]
源码 里 就有啊 我都是用源码和别人编译好的

**Reply by 594810899** [reply]
> 源码 里 就有啊 我都是用源码和别人编译好的


方便加一下你QQ吗？有些不懂的地方想向你请教一下 我Q:594810899

**Reply by 594810899** [reply]
> 源码 里 就有啊 我都是用源码和别人编译好的

想请问一下 我这边设置好了参数后  连接成功 但是上不了外网，是不是还需要设置哪里？

**Reply by 594810899** [reply]
> 源码 里 就有啊 我都是用源码和别人编译好的
帮忙看一下 我的 conf是这样写的，有没有问题？

let conf = """
[General]
loglevel = trace
dns-server = 223.5.5.5, 114.114.114.114, 8.8.8.8
tun-fd = REPLACE-ME-WITH-THE-FD
routing-domain-resolve = true

[Proxy]
Direct = direct
Proxy = trojan, 212.50.244.81, 443, password=****
[Rule]
EXTERNAL, site:cn, Direct
EXTERNAL, mmdb:cn, Direct
FINAL, Proxy
"""

**Comment by Chen77777** [comment]
我也有此问题、由于连接账号密码和规则都需要动态配置、考虑到使用json这种格式、否则字符串直接的拼接需要遵循严格的排版比较容易出错
